src/routes/notifications.rs

```diff
@@ -0,0 +1,65 @@
+use crate::utils::context::Context;
+
+use crate::models::service::notifications::NotificationRegistrationRequest;
+use crate::services::notifications::{delete_registration, post_registration};
+use crate::utils::errors::ApiResult;
+use rocket::serde::json::Error;
+use rocket::serde::json::Json;
+
+/**
+ * `/v1/register/notifications` <br />
+ * Returns `()`
+ *
+ * # Register notifications
+ *
+ * This endpoint provides a way for registering devices for push notifications.
+ *
+ * One can subscribe to as many safes in different chains as [SafeRegistration](crate::models::service::notifications::SafeRegistration) provided in the payload
+ *
+ * ## Path
+ *
+ * `POST /v1/register/notifications`
+ *
+ * ## Query parameters
+ *
+ * No query parameters available for this endpoint.
+ */
+#[post(
+    "/v1/register/notifications",
+    format = "application/json",
+    data = "<registration_request>"
+)]
+pub async fn post_notification_registration<'e>(
+    context: Context<'_>,
+    registration_request: Result<Json<NotificationRegistrationRequest>, Error<'e>>,
```

Comment:
Why do we require a `Result` here? (e.g. in the hook we directly have the `Json`, while in the confirmation screen we also use a result) Is this for better error generation?

Reply:
Yeah, better errors from rocket :+1: